cpp/src/Ice/SSL/OpenSSLTransceiverI.cpp

```diff
@@ -110,6 +110,13 @@ OpenSSL::TransceiverI::initialize(IceInternal::Buffer& readBuffer, IceInternal::
 
         SSL_set_ex_data(_ssl, 0, this);
         SSL_set_verify(_ssl, SSL_get_verify_mode(_ssl), Ice_SSL_opensslVerifyCallback);
+        // Enable SNI by default for outgoing connections. The SNI host name is always empty for incoming connections.
+        if (!_host.empty() && !SSL_set_tlsext_host_name(_ssl, _host.c_str()))
```

Comment:
For my understanding: _host here can be either a DNS name or an IP address? And can't be empty for an outgoing connection?

Reply:
The host here come from IPEndpointInfo and can be empty for outgoing connections. The endpoint host resolver handles empty host explicit and resolves it to the loopback address.

https://github.com/zeroc-ice/ice/blob/652cf90d8f36192e5f7d9be75d74f227083399df/cpp/src/Ice/IPEndpointI.cpp#L530

https://github.com/zeroc-ice/ice/blob/652cf90d8f36192e5f7d9be75d74f227083399df/cpp/src/Ice/Network.cpp#L810-L815